### DISCUSSION TITLE
Can Basana be used for paper trading like pyalgotrade?

### DISCUSSION THREAD

**Discussion by NagarajuGunda** [gbeced/basana#22] [Q&A] [answered]
I have started using pyalgotrade as base for my trading with my code on top of it(https://github.com/NagarajuGunda/PyAlgoMate) to support indian stock markets

I am a big fan of pyalgotrade and use backtesting, paper trading and live trading using it

I am considering moving to async based basana, Just wanted to check if all that is possible using pyalgotrade is possible using basana?

**Comment by gbeced** [comment, marked as answer]
No. Basana is, by design, a smaller package. As explained in the docs, the framework has 3 main components:

* The core, where basic abstractions like events, event sources and the event dispatcher live.
* A backtesting exchange that you can use to validate your strategies before using real money.
* External integrations, where you’ll find support for live trading at [Binance](https://www.binance.com/) and [Bitstamp](https://www.bitstamp.net/) crypto currency exchanges.

A couple of differences with PyAlgoTrade:
* Python 3 only.
* Basana doesn't ship with technical indicators. There are many packages out there and it didn't make sense to me to redo all that work. I use [talipp](https://github.com/nardew/talipp) in the examples.
* Right now there is no support for metrics like drawdown, sharpe, etc. These will be available in the future though.
* The position based interface (enterLong, enterShort, etc) doesn't exist in Basana.
